port/linux/socklistener.h

```diff
@@ -0,0 +1,53 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef SOCKLISTENER_H
+#define SOCKLISTENER_H
+
+#include <stdint.h>
+#include <sys/socket.h>
+
+typedef struct oc_sock_listener_t
+{
+  struct sockaddr_storage sockaddr;
+  int sock;
+  uint16_t port;
+} oc_sock_listener_t;
+
+#define OC_CLOSE_SOCK_LISTENER(server)                                         \
```

Comment:
Why the use of macros instead of functions?

clang-tidy complains about stuff inside, see: 
https://github.com/iotivity/iotivity-lite/actions/runs/4980635641/jobs/8913797112

look for port/linux/socklistener.h

Reply:
fixed